### ISSUE TITLE
Work boats cannot enter ocean tiles within you culture borders until Astronomy is researched

### ISSUE THREAD

**Issue by Kzer-Za** [yairm210/Civ-IV#87]
Game version: 4.14.9.
Mod info: updated 2024-11-25.
OS: Android.

This may be not a bug and I apologize if it's not. I have no copy of the actual Civ IV to check its behavior. But in https://civilization.fandom.com/wiki/Work_Boat_(Civ4) it says that "The Work Boat ... cannot enter ocean spaces outside one's own cultural borders until Astronomy has been researched. "

Judging by the phrasing, I would say that *within* your cultural borders work boats *should* be able to enter ocean tiles even before Astronomy. Which would seem logical, because Whales are encountered only on ocean tiles and are obsolete with Combustion. So if work boats cannot (even within your cultural borders) enter ocean tiles until Astronomy, it would mean that the window when you can use this resource is *very* slim.

Prerequisites:

1. Have a city whose culture borders expand into an ocean tile, which is thus within your cultural borders.
2. Have a work boat unit built and located on a coast tile adjacent to the aforementioned ocean tile.

Steps to reproduce:

1. Select the work boat.
2. Try to move it to the aforementioned ocean tile.

The expected result: the work boat should move to the aforementioned ocean tile.

The actual result: the work boat can't be moved there.

**Comment by RobLoach** [comment]
Hmmm, yeah, allowing it to be in nation culture border, in ocean before Astronomy may not be possible. For now, I fixed the case for Astronomy instead of Optics in `"Cannot enter ocean tiles <before discovering [Astronomy]>"`.

**Comment by Kzer-Za** [comment]
Could you maybe fix it so they "Cannot enter ocean tiles before discovering Optics"? Since in Civ IV they can enter those tiles from the beginning (even only within your cultural borders), requiring Astronomy seems an overkill: it completely prevents you from working ocean tiles until practically mid-game.

Of course, since its impossible to implement the limitation of them being able to enter ocean tiles only within the borders, it could be said, that giving the work boats this ability should be delayed until a proper tech is researched.

But I think that Optics is a good middle ground: work boats can't enter ocean tiles (within the borders) from the beginning of the game, but they can enter them even outside the borders once you have Optics. While delaying this ability until Astronomy, like I said, completely prevents you from working ocean until the middle of the Renaissance, which is rather harsh on the whole, but particularly crippling in regard of Whales. Whales are obsolete with Combustion, and Combustion is not all that far turn-wise from Astronomy, which makes this resource almost unusable.

**Comment by RobLoach** [comment]
Good call. Pushed up `"Cannot enter ocean tiles <before discovering [Optics]>"`
